source/common/common/regex.cc

```diff
@@ -22,6 +22,11 @@ class CompiledStdMatcher : public CompiledMatcher {
     return std::regex_match(value.begin(), value.end(), regex_);
   }
 
+  // CompiledMatcher
+  std::string replaceAll(absl::string_view value, absl::string_view substitution) const override {
+    return std::regex_replace(std::string(value), regex_, std::string(substitution));
```

Comment:
I don't expect this to be used by your feature (since only re2 is supported there), but `std::regex_replace` may throw. I think we should probably handle that here and return the original string.

Reply:
Added with the latest commit.